### ISSUE TITLE
read_parquet pyarrow memory build up

### ISSUE THREAD

**Issue by mgsnuno** [dask/dask#5563]
create dummy dataframe (using `datasets.timeseries()` and mapping all elements with `sha256`)
```python
import hashlib
import dask
import dask.dataframe as dd
from dask.distributed import Client

client = Client()
path = "./tmp/parquet_test/"

timeseries = dask.datasets.timeseries().reset_index()
timeseries = timeseries.applymap(
    lambda x: hashlib.sha256(str(x).encode("utf8")).hexdigest(), meta=timeseries
).astype(str)

timeseries.to_parquet(path, engine="fastparquet")
```

multiple reads using `fastparquet`: `Bytes Stored` remains constant
```python
timeseries = dd.read_parquet(path, engine="fastparquet")
for _ in range(100):
    _ = timeseries.head()
```

multiple reads using `pyarrow`: `Bytes Stored` builds up
```python
timeseries = dd.read_parquet(path, engine="pyarrow")
for _ in range(100):
    _ = timeseries.head()
```

`pyarrow` should not cause this memory build up. Any clues and should I leave this here or post it in arrow?

Thank you

**Comment by mrocklin** [comment]
Thank you for the excellent bug report @mgsnuno .  

I think that the next step here is probably to try to isolate the problem further.  If you have time, I recommend seeing if you can recreate the error without Dask, using only Pandas and Arrow.  If you can accomplish this, then yes, I would recommend posting this to the Arrow issue tracker.  They will be much better at diagnosing issues there than we are.  

I recommend the following experiments:

1.  Try without threading with `timeseries.head(compute=False).compute(scheduler='single-threaded')` and see if the memory leak continues.  Often memory management systems don't interact well with multi-threaded applications.
2.  If the threading experiment above shows that we need threads to create the memory leak,  then try `pandas.read_parquet(...)` in many threads, perhaps using the `concurrent.futures.ThreadPoolExecutor.map` method.  If the memory leak occurs without threads, then try the same but with a normal for loop.

**Comment by mgsnuno** [comment]
Issue fixed with latest pyarrow 0.15.1. Closing

Here the related issues (imo) that fixed it: 
https://issues.apache.org/jira/browse/ARROW-6060
https://github.com/apache/arrow/pull/5016

**Comment by mrocklin** [comment]
Thanks for checking back in and reporting the solution here!
